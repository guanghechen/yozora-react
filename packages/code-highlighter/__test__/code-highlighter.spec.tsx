import React, { useEffect, useState } from 'react'
import { mount, render } from 'enzyme'
import CodeHighlighter from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('test lineno change', () => {
    function Wrapper() {
      const [code, setCode] = useState<string>('let a: number = 1 + 2;')
      const [lineCount, setLineCount] = useState<number>(0)
      const lineNoWidth = `${ Math.max(2, ('' + lineCount).length) + 0.5 }em`

      useEffect(() => {
        const nextCode = (
          'let a = 1, b = 2\n' +
          Array.from(new Array(100)).map((x, i) => '// ' + i).join('\n') +
          '\nlet c = 3\nconsole.log(\'c:\', c)'
        )
        setCode(nextCode)
      }, [])

      return (
        <pre data-line-count={ lineCount }>
          <CodeHighlighter
            lang="typescript"
            value={ code }
            linenoWidth={ lineNoWidth }
            onLineCountChange={ setLineCount }
          />
        </pre>
      )
    }

    const wrapper = mount(<Wrapper />)
    expect(wrapper.getDOMNode().getAttribute('data-line-count')).toEqual(String(103))

    const lines = wrapper.find({ linenoWidth: '2.5em' })
    for (let i = 0; i < lines.length; ++i) {
      const line = lines[i]
      const s = getComputedStyle(line.getDOMNode())
      expect(s.width).toEqual('3.5em')
    }
  })

  it('snapshot', () => {
    const wrapper = render(
      <pre>
        <CodeHighlighter
          lang="typescript"
          value="let a: number = 1 + 2;"
          linenoWidth="2em"
        />
      </pre>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
