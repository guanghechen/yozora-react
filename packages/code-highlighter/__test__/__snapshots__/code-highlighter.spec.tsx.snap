// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c0 {
  text-indent: -2em;
  padding-left: 2em;
}

.c1 {
  display: inline-block;
  width: 2em;
  padding-right: 0.6em;
  color: #858585;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: right;
}

<pre>
  <div
    class="c0 token-line"
    style="color:#9CDCFE"
  >
    <span
      class="c1"
    >
      1
    </span>
    <span
      class="token keyword"
      style="color:rgb(86, 156, 214)"
    >
      let
    </span>
    <span
      class="token plain"
    >
       a
    </span>
    <span
      class="token punctuation"
      style="color:rgb(212, 212, 212)"
    >
      :
    </span>
    <span
      class="token plain"
    >
       
    </span>
    <span
      class="token builtin"
      style="color:rgb(86, 156, 214)"
    >
      number
    </span>
    <span
      class="token plain"
    >
       
    </span>
    <span
      class="token operator"
      style="color:rgb(212, 212, 212)"
    >
      =
    </span>
    <span
      class="token plain"
    >
       
    </span>
    <span
      class="token number"
      style="color:rgb(181, 206, 168)"
    >
      1
    </span>
    <span
      class="token plain"
    >
       
    </span>
    <span
      class="token operator"
      style="color:rgb(212, 212, 212)"
    >
      +
    </span>
    <span
      class="token plain"
    >
       
    </span>
    <span
      class="token number"
      style="color:rgb(181, 206, 168)"
    >
      2
    </span>
    <span
      class="token punctuation"
      style="color:rgb(212, 212, 212)"
    >
      ;
    </span>
  </div>
</pre>
`;
